1-foundations/2-control-flows-and-functions/foundations-s2.ipynb

```diff
@@ -1314,38 +1497,82 @@
     "\n",
     "def query_population(country = 'AFG', series = \"SP.POP.TOTL\", time = 2020):\n",
     "    \"\"\"\n",
-    "    Function to retrieve indicator level data from WDI database. \n",
+    "    Function to retrieve population data from WDI database. \n",
     "    \n",
     "    Inputs:\n",
-    "    \n",
+    "\n",
     "    country : ISO-3 code for country\n",
     "    series : Indicator code\n",
     "    time : year you want to query data for\n",
     "    \n",
-    "    Returns: Population (in millions)\n",
+    "    Returns: \n",
+    "    \n",
+    "    Population (in millions)\n",
     "    \"\"\"\n",
     "    import requests\n",
     "    \n",
+    "    ## For this session, you do not need to know how APIs work\n",
+    "    ## What you need to know it, the variable `req` has a dictionary object stored\n",
+    "    ## We need to parse through that dictionary object to access the data we want\n",
+    "    \n",
     "    req = requests.get(\"https://api.worldbank.org/v2/sources/2/country/{}/series/{}/time/{}?format=json\".format(country, series, 'yr'+str(time))).json()\n",
     "    \n",
     "    population = req['source']['data'][0]['value'] / 1e6\n",
     "    \n",
-    "    return population"
+    "    return round(population,3)"
    ]
   },
   {
+   "cell_type": "code",
+   "execution_count": 44,
```

Comment:
Suggesting adding a markdown cell above, maybe with an image, to explain what an API is. 

Then flip order of the two cells. So you first show how this can be done without the function and explain each step, and then show how one can make it easier for oneself by encapsulate those details in a function.